include/renderer_gl/surfaces.hpp

```diff
@@ -29,12 +33,32 @@ struct ColourBuffer {
         range = Interval<u32>(loc, (u32)endLoc);
     }
 
-    void allocate() {
+	void allocate() {
+		// Internal formats for the texture based on format
+		static constexpr std::array<GLenum, 5> internalFormats = {
+			GL_RGBA8, GL_RGB8, GL_RGB5_A1, GL_RGB565, GL_RGBA4
+		};
+
+		// Format of the texture
+		static constexpr std::array<GLenum, 5> formats = {
+			GL_RGBA, GL_BGR, GL_RGBA, GL_RGB, GL_RGBA,
+		};
+
+		static constexpr std::array<GLenum, 5> types = {
+			GL_UNSIGNED_INT_8_8_8_8, GL_UNSIGNED_BYTE, GL_UNSIGNED_SHORT_5_5_5_1,
+			GL_UNSIGNED_SHORT_5_6_5, GL_UNSIGNED_SHORT_4_4_4_4,
+		};
+
+		internalFormat = internalFormats[(int)format];
+		fmt = formats[(int)format];
+		type = types[(int)format];
+
+
         // Create texture for the FBO, setting up filters and the like
         // Reading back the current texture is slow, but allocate calls should be few and far between.
         // If this becomes a bottleneck, we can fix it semi-easily
         auto prevTexture = OpenGL::getTex2D();
-        texture.create(size.x(), size.y(), GL_RGBA8);
+        texture.create(size.x(), size.y(), internalFormat);
```

Comment:
Won't this end up degrading image quality? I initially used GL_RGBA8 here always hoping that the extended precision would make for nicer textures.

Reply:
It's a question if you want to be more accurate to the hardware or better image quality. I haven't really tested if this affects image quality, though having the correct format makes texture downloads simpler. Also in vulkan if you have the correct format you can do reinterpretation with a simple image copy instead of shader in some cases.

Reply:
Reverting temporarily